### ISSUE TITLE
I wanted to display filter panel only under selected column

### ISSUE THREAD

**Issue by ramusesan** [mui/mui-x#9949]
### Order ID

69192

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### The problem in depth 🔍

when I selected a filter from any column, the filter panel is always displayed at the beginning (left-hand side).

![image](https://github.com/mui/mui-x/assets/15937743/9e5f674b-8f92-46cc-b001-1a81f0554cc8)

but I wanted this panel to be displayed under the selected column for example. if I selected filter from the Phone column, this filter panel should be displayed under the phone- column not at the beginning. and whatever column is chosen for the filter? 

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by joserodolfofreitas** [comment]
Thanks for pointing this out @ramusesan; I think this is a good aspect to consider for #6419 as well. 
I wondered if [filters on headers](https://mui.com/x/react-data-grid/filtering/header-filters/) could be a good UX for your use case. Have you considered it?

**Comment by ramusesan** [comment]
yes this helping me thanks for your reply, bit still I feel it would be good if there is a prop to enable to display of the filter under selected column
